Deb: Sync Salsa-CI from Debian MariaDB 10.5 repository

ottok opened the pull request:
Since Debian Sid now has MariaDB 10.6, we can't do upgrade tests on the
10.5 branch in Sid anymore, it just fails with downgrade errors.

Also, since MariaDB 10.5 is no longer in Sid, we can't even test 10.5.x
to 10.5.y upgrades in Sid.

Instead the 10.5 branch salsa-ci.yml should run all builds and tests based
on Debian Bullseye, which has MariaDB 10.5 (only).

To achieve this, essentially sync most the the salsa-ci.yml contents from
https://salsa.debian.org/mariadb-team/mariadb-10.5/-/tree/bullseye

Also add a couple Lintian overrides to make Salsa-CI pass.

**NOTE TO MERGERS: This commit is intended for the 10.5 branch only, do not
merge anything from it on 10.6 or any other branch.**

## Basing the PR against the correct MariaDB version
- [x] *This is a bug fix and the PR is based against the earliest branch in which the bug can be reproduced*

## Testing

See CI result at https://salsa.debian.org/mariadb-team/mariadb-server/-/pipelines/354230 (failure in autopkgtest is unrelated):
![image](https://user-images.githubusercontent.com/668724/156522258-cedf0b65-b86a-4cf3-9f3b-330c619edd1f.png)

illuusio commented on debian/salsa-ci.yml:
```diff
@@ -46,10 +46,9 @@ build:
+# build jobs often needs more than default GitLab timeout (1h)
+.build-package:
+  timeout: 3h
 
```
Should Gitlab runtime timeout used as I don't know does this even work?

ottok replied:
https://salsa.debian.org/mariadb-team/mariadb-server/-/commit/7abb3a4f813ae6032dc9bcce40d0a7c00df6c50d
  GitLab CI now supports timeout for specific jobs

illuusio commented on debian/salsa-ci.yml:
```diff
@@ -199,34 +198,8 @@ mariadb-10.5 Sid upgrade:
       - $CI_COMMIT_TAG != null && $SALSA_CI_ENABLE_PIPELINE_ON_TAGS !~ /^(1|yes|true)$/
 
-mariadb-10.5 Bullseye to mariadb-10.5 upgrade:
-  stage: upgrade from Bullseye/Buster/Stretch
```
Why this is removed?

illuusio commented on debian/salsa-ci.yml:
```diff
@@ -493,42 +431,12 @@ mariadb-connector-c Stretch upgrade:
-# Upgrading from MySQL 8.0 with datadir in place is not possible. Users need to do a data dump.
-# The Debian maintainer scripts detect this situation and simply moves old datadir aside and start fresh.
-mysql-8.0 Sid to mariadb-10.5 upgrade:
-  stage: upgrade in Sid
```
Why this is removed? Is MySQL upgrade malfunctioning?

illuusio commented on debian/salsa-ci.yml:
```diff
@@ -614,14 +525,14 @@ mariadb.org-10.5 to mariadb-10.5 upgrade:
     - apt install -y curl
     - curl -sS https://mariadb.org/mariadb_release_signing_key.asc -o /etc/apt/trusted.gpg.d/mariadb.asc
-    - echo 'deb http://mirror.one.com/mariadb/repo/10.5/debian sid main' > /etc/apt/sources.list.d/mariadb.list
+    - echo 'deb http://mirror.one.com/mariadb/repo/10.5/debian bullseye main' > /etc/apt/sources.list.d/mariadb.list
```
This should be same URL than in 10.6.

ottok replied:
Updated

illuusio commented on debian/salsa-ci.yml:
```diff
@@ -614,14 +525,14 @@ mariadb.org-10.5 to mariadb-10.5 upgrade:
     - apt install -y curl
     - curl -sS https://mariadb.org/mariadb_release_signing_key.asc -o /etc/apt/trusted.gpg.d/mariadb.asc
-    - echo 'deb http://mirror.one.com/mariadb/repo/10.5/debian sid main' > /etc/apt/sources.list.d/mariadb.list
+    - echo 'deb http://mirror.one.com/mariadb/repo/10.5/debian bullseye main' > /etc/apt/sources.list.d/mariadb.list
```
This URL should be upgraded to same as in 10.6

illuusio commented on debian/salsa-ci.yml:
```diff
@@ -493,42 +431,12 @@ mariadb-connector-c Stretch upgrade:
-
-# Upgrading from MySQL 8.0 with datadir in place is not possible. Users need to do a data dump.
-# The Debian maintainer scripts detect this situation and simply moves old datadir aside and start fresh.
-mysql-8.0 Sid to mariadb-10.5 upgrade:
```
Is this malfunctioning or why this test is removed?

ottok replied:
Since Debian Sid now has MariaDB 10.6, we can't do any upgrade tests in
Debian Sid for the 10.5 branch anymore. It would just fail with downgrade
errors.

illuusio commented on debian/salsa-ci.yml:
```diff
@@ -383,41 +355,8 @@ build mariadbclient consumer Python-MySQLdb:
     variables:
       - $CI_COMMIT_TAG != null && $SALSA_CI_ENABLE_PIPELINE_ON_TAGS !~ /^(1|yes|true)$/
 
-libmysql* to libmariadb* upgrade:
```
Is this not needed to be tested anymore?

ottok replied:
Since Debian Sid now has MariaDB 10.6, we can't do any upgrade tests in
Debian Sid for the 10.5 branch anymore. It would just fail with downgrade
errors.

ottok replied:
MySQL in Debian exists only in Sid, we can't use other Debian releases for these tests either.

illuusio commented on debian/salsa-ci.yml:
```diff
@@ -638,111 +549,6 @@ mariadb.org-10.5 to mariadb-10.5 upgrade:
     variables:
       - $CI_COMMIT_TAG != null && $SALSA_CI_ENABLE_PIPELINE_ON_TAGS !~ /^(1|yes|true)$/
 
-mariadb.org-10.4 to mariadb-10.5 upgrade:
```
Are these duplicates or why these are removed?

ottok replied:
Updated version pushed to https://salsa.debian.org/mariadb-team/mariadb-server/-/pipelines/356697 - will check tomorrow if it makes sense to have these old upgrade tests.

ottok commented:
@illuusio See [git blame of original Debian file](https://salsa.debian.org/mariadb-team/mariadb-server/-/blame/debian/latest/debian/salsa-ci.yml) for information about what the original commits where. In this PR I decided to just sync the file in one commit and not import each commit as a separate one.

ottok commented:
@illuusio I have addressed all review feedback. Thanks for reading the commit in detail. Please re-review now.
Salsa-CI results visible at https://salsa.debian.org/mariadb-team/mariadb-server/-/pipelines/358468
![image](https://user-images.githubusercontent.com/668724/158314446-b1047d34-3309-43c0-befe-996cf7ae803d.png)

illuusio commented on debian/salsa-ci.yml:
```diff
@@ -103,24 +102,18 @@ blhc:
-  echo 'deb http://deb.debian.org/debian sid main' > /etc/apt/sources.list
+.test-enable-bullseye-repos: &test-enable-bullseye-repos |
+  # Replace any old repos with just Bullseye
+  echo 'deb http://deb.debian.org/debian bullseye main' > /etc/apt/sources.list
```
Please use '"' as everything else is converted to use it

illuusio replied:
Should this use ${RELEASE} not bullseye?

ottok replied:
Using now `"` everywhere in deb lines. In the repo change `${RELEASE}` can't be used, as it needs to control exactly what distro release the upgrade test is going into.

illuusio commented on debian/salsa-ci.yml:
```diff
@@ -685,60 +591,73 @@ mariadb.org-10.3 to mariadb-10.5 upgrade:
-    - mysql --defaults-file=/etc/mysql/debian.cnf -e "SELECT * FROM plugin;" mysql
+    - *test-verify-initial
+    # Enable backports to make galera-4 available
+    - echo 'deb http://deb.debian.org/debian buster-backports main' >> /etc/apt/sources.list.d/backports.list && apt-get update
```
Please use '"' as everywhere else even though there is not variable to extend.

illuusio reviewed (commented):
Just minor things otherwise LGTM

ottok commented:
Latest commits on the 10.5 branch already broke the CI. If this would have been merged yesterday or before those changes, we would have had a chance to detect them automatically. Merging this now myself to prevent further unnecessary delays as this is already Approved anyway.
